pkg/cloudprovider/cloudprovider.go

```diff
@@ -337,3 +338,22 @@ func ParseResourceID(resourceID string) (azure.Resource, error) {
 
 	return result, nil
 }
+
+// recordError records the error in appropriate metric
+func recordError(reporter *metrics.Reporter, operation string) {
```

Comment:
Same here. please.

Reply:
Whatever approach we freeze in above, will be followed here.